aspnet/dnx/projects.rst

```diff
@@ -1,8 +1,652 @@
+
+Restoring packages
+------------------
+
+The DNX Utility (DNU) wraps the functionality of NuGet to do package restore, which means that it uses the *NuGet.config* file to determine where to download the package. If you want to get packages from somewhere other than NuGet.org, you can edit your NuGet.config.
+
+For instructions about how to get development builds of the latest ASP.NET and DNX packages, see `Configuring the feed used by dnu to restore packages <https://github.com/aspnet/Home/wiki/Configuring-the-feed-used-by-dnu-to-restore-packages>`_.
+
+Using DNU Restore
+^^^^^^^^^^^^^^^^^
+
+The restore command will look at the dependencies listed in the *project.json* file and download them, adding them to your app's packages directory. It downloads the entire graph of dependencies, even though you only explicitly declare the top level dependency that you directly require. It uses NuGet internally to download packages. 
+
+The following is an example of using the ``restore`` command from the command window, where the command is executed from the folder containing the application (including the *project.json* file)::
+
+	dnu restore
+
+Project lock file
+^^^^^^^^^^^^^^^^^
+
+When doing a package restore, DNU builds up a great deal of information about the dependencies of your application, this information is persisted to disk in the *project.lock.json* file.
+
+The DNX reads the lock file when running your application instead of rebuilding all the information that the DNU already generated. To understand the reason for that, imagine what the DNX has to do without the lock file:
+
+1. Find each dependency listed in the *project.json* file.
+2. Open the nuspec of each package and get all of their dependencies.
+3. Repeat step 2 for each dependency until it has the entire graph.
+4. Load all the dependencies.
+
+By using the lock file, this process is reduced to:
+
+1. Read the lock file.
+2. Load all the dependencies.
+
+There is significantly less disk IO involved in the second list.
+
+.. Note:: Because it has such a large impact on the cold start performance of DNX applications, the lock file is mandatory to run. If you do not have a lock file, the DNX will fail to load your application.
+
+In addition to this, the DNU also does things like determine if the package should be bread-crumbed for servicing and marks it as such in the lock file. See :doc:`Servicing DNX Applications <servicing>`.
+
+The lock file ensures that after you run ``dnu restore``, you have a fixed set of packages that you are referencing. When restoring, the DNU generates the lock file which specifies the exact versions that your project will use. This way, versions only get modified when you run ``dnu restore``, not during run-time. Restoring also ends up improving performance at run-time since the DNX no longer has to probe the packages directory to find the right version to use, the DNX just does what the lock file instructs the DNX to do. 
```

Comment:
Also, the last sentence on performance seems redundant with the earlier discussion of performance (which could now be moved down here)